tools/python_api/src_cpp/py_connection.cpp

```diff
@@ -404,6 +347,28 @@ static LogicalType pyLogicalType(const py::handle& val) {
         return LogicalType::INTERVAL();
     } else if (py::isinstance(val, uuid)) {
         return LogicalType::UUID();
+    } else if (py::isinstance<py::dict>(val)) {
```

Comment:
I am a bit confused, what if the user wants to interpret the py dict as struct instead of map?

Reply:
It depends. If they're using prepared statement parameters, we go through pyLogicalTypeFromParameter, which interprets the py::dict as a struct. If they're using UDF, specifying the return type bypasses pyLogicalType entirely, and the code assumes their dict is a struct.

Reply:
tl;dr users shouldn't notice a difference from this PR